### PR TITLE
chore: create tenderdash role for seed nodes

### PR DESCRIPTION
**Pull request by strophy**
## Issue being fixed or feature implemented

Seed nodes should no longer be deployed by dashmate

## What was done?

Implement Tenderdash-only role in deploy tool

## How Has This Been Tested?

On testnet seed-3

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by ktechmidas — APPROVED** [review]
LGTM

File: ansible/roles/tenderdash/templates/docker-compose.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: '3'
+
+services:
+  tenderdash:
+    image: {{ tendermint_image }}
+    container_name: tenderdash
+    entrypoint: sh -c "/usr/bin/tenderdash node"
```
Comment (shumkov) [inline ansible/roles/tenderdash/templates/docker-compose.yml.j2:7]: Why do we need this?

File: ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (shumkov) [inline ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2:1]: Seed nodes need this?

File: ansible/roles/tenderdash/templates/tenderdash/genesis.json.j2 [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "genesis_time": "{{ genesis_time }}",
+  "chain_id": "dash-{{ ( 'devnet-' + dash_devnet_name if dash_network == 'devnet' else dash_network ) if tenderdash_chain_id is not defined else tenderdash_chain_id }}",
+  "consensus_params": {
+    "timeout": {
```
Comment (shumkov) [inline ansible/roles/tenderdash/templates/tenderdash/genesis.json.j2:5]: Initial core chain lock height missed?

File: ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2 (hunk 2 above)
Comment (strophy) [reply ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2:1]: I don't know, I just combined the dashmate/mn_evo_services deploy methods to achieve this

File: ansible/roles/tenderdash/templates/docker-compose.yml.j2 (hunk 1 above)
Comment (strophy) [reply ansible/roles/tenderdash/templates/docker-compose.yml.j2:7]: I don't know, I just replicated the old config method

File: ansible/roles/tenderdash/templates/tenderdash/genesis.json.j2 (hunk 3 above)
Comment (strophy) [reply ansible/roles/tenderdash/templates/tenderdash/genesis.json.j2:5]: Good catch, although it doesn't seem to be necessary?

File: ansible/roles/tenderdash/templates/docker-compose.yml.j2 (hunk 1 above)
Comment (shumkov) [reply ansible/roles/tenderdash/templates/docker-compose.yml.j2:7]: We don't have such thing in dashmate. We don't need this anymore.

File: ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2 (hunk 2 above)
Comment (shumkov) [reply ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2:1]: This one is needed for validators only. Let's try without it?

File: ansible/roles/tenderdash/templates/tenderdash/genesis.json.j2 (hunk 3 above)
Comment (shumkov) [reply ansible/roles/tenderdash/templates/tenderdash/genesis.json.j2:5]: I'm not sure we need something except chain_id and genesis time for seed nodes at all

Comment (strophy) [reply ansible/roles/tenderdash/templates/tenderdash/genesis.json.j2:5]: `validator_quorum_type` is needed as well, everything else successfully removed

File: ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2 (hunk 2 above)
Comment (strophy) [reply ansible/roles/tenderdash/templates/tenderdash/node_key.json.j2:1]: Works fine without it

**Review by shumkov — APPROVED** [review]
👍

File: ansible/roles/kill_platform/tasks/main.yml [hunk 4]
```diff
@@ -4,7 +4,21 @@
...
+  when: inventory_hostname in groups["hp_masternodes"]
+
+- name: Remove platform (seeds)
+  community.docker.docker_compose:
+    project_src: '{{ tenderdash_compose_path }}'
+    state: absent
+  when:
+    - inventory_hostname not in groups["hp_masternodes"]
+
+- name: Delete platform dir (seeds)
```
Comment (shumkov) [inline ansible/roles/kill_platform/tasks/main.yml:16]: Delete tenderdash dir (seeds)

File: ansible/roles/kill_platform/tasks/main.yml [hunk 5]
```diff
@@ -4,7 +4,21 @@
   ansible.builtin.import_role:
     name: dashmate
     tasks_from: destroy_platform
-  when: inventory_hostname in groups["hp_masternodes"] or inventory_hostname in groups["seed_nodes"]
+  when: inventory_hostname in groups["hp_masternodes"]
+
+- name: Remove platform (seeds)
```
Comment (shumkov) [inline ansible/roles/kill_platform/tasks/main.yml:9]: Remove tenderdash (seeds)

**Review by shumkov — CHANGES_REQUESTED** [review]
Let's use cheaper instances for seeds?

**Comment by strophy** [comment]
OK, good idea. Please approve and I'll deploy this change tomorrow (will need to slowly restart all nodes with new seed IPs in config)

**Pull request merged**